### PR TITLE
Fixes oil slime reactions in grenades

### PR DESCRIPTION
**Pull request by KoboldCommando**
# About The Pull Request

For some reason oil slime cores in large grenades would not explode.  This ruined the plans of a player during a round.  I've retooled the proc so it will explode properly, similar to how I fixed black powder grenades.

## Why It's Good For The Game

Thing working as intended is good

## Testing Photographs and Procedure

In a large grenade casing, add an oil slime core and a beaker of plasma.  Trigger, observe that it does not explode properly.  With this PR, observer that it will now blow your ass up.

image

## Changelog

:cl:
fix: oil slime cores in grenades should now explode properly!
/:cl:

### CODE REVIEW

**Comment by KoboldCommando** [comment]
> 

Unsure.  I wanted to prioritize fixing this for our server.  I may reorganize this and the black powder problem as an upstream PR later.  Right now others have said that the grenade does work on other codebases so I tackled it under that assumption, I haven't tested myself.

**Comment by nednaZ** [comment]
Currently not working as intended, slime core will detonate user when not in a grenade casing.

**Comment by KoboldCommando** [comment]
Retooled it all to take a different approach shamelessly taken from tgstation.  Both grenades and raw cores work in my testing.

**Comment by nednaZ** [comment]
Requires fixes for oversaturation in grenades before merge

**Comment by KoboldCommando** [comment]
Grenades should no longer runtime.  In addition, if >100u of chems are injected (meaning some is left behind in the beaker) it should now work fine, splashing the remaining contents on the floor.  A grenade should drop empty beakers and non-beaker items on the floor, but beakers with reagents should be destroyed.

File: monkestation/code/game/objects/items/storage/boxes.dm [hunk 1]
```diff
@@ -0,0 +1,18 @@
+/*
+ *	Pre-packed boxes for testing purposes
+ */
+
+/obj/item/storage/box/oilgrenade
```
Comment (nednaZ) [inline monkestation/code/game/objects/items/storage/boxes.dm:5]: This file doesn't appear to be in the .dme
May want to remove it if it's no longer needed for testing.

**Review by nednaZ — COMMENTED** [review]
Appears to be functional without any runtimes, regular grenades unaffected.
Just remove the spare file and it should be ready.

**Pull request merged**